src/cpp/src/sampler.hpp

```diff
@@ -105,7 +105,7 @@ class Sampler::GroupBeamSearcher {
         bool done = false;
 
         int64_t finish(Beam beam, const ov::genai::GenerationConfig& sampling_params);
-        void is_done(const ov::genai::GenerationConfig& sampling_params);
+        void is_done(const ov::genai::GenerationConfig& sampling_params, size_t prompt_len);
```

Comment:
I think we can include `prompt_len` information in `Group` members during object construction and avoid passing it as a parameter in this method.